### ISSUE TITLE
Document the proper method for using VSCode Remote on ShARC

### ISSUE THREAD

**Issue by jkwmoore** [rcgsheffield/sheffield_hpc#1196]
To users: please stop using VSCode on the login node! You are breaking your own envs and saturating cores!

Now that is said! By editing the settings.json in VSCode we can get it to spawn an interactive session.

See: C:\Users\USERNAME\AppData\Roaming\Code\User\settings.json or %APPDATA%\Code\User\settings.json

File: 
```
{
    "remote.SSH.remotePlatform": {
        "sharc.shef.ac.uk": "linux",
    },
    "terminal.integrated.shellArgs.linux": [ "-c","export FAF=FEF ; exec qrshx -V",]
}
```
Note: Further resource requests e.g. RAM may be required for certain workloads.

![image](https://user-images.githubusercontent.com/69149348/116101885-21fddf80-a6a6-11eb-8e08-a66e6d1fa566.png)

**Comment by jkwmoore** [comment]
Pinning this here so I don't forget: https://github.com/microsoft/vscode-remote-release/issues/1722#issuecomment-718246436

**Comment by IanSudbery** [comment]
I tried this. As far as I can tell its still running processes on the head node. In fact, it appears to be completely ignoring the changes to settings.json as terminal sessions still start on the head node, and show no interactive bash job on qstat.

**Comment by jkwmoore** [comment]
Yes, at the moment it is subject to the same conditions as normal qrshx sessions - if the scheduler is unable to provide you immediately with an interactive session it will drop onto the login nodes and continue there instead.

We may need to look into a specific spawning command which will terminate the session fully if an interactive session is not acquired.

**Comment by jkwmoore** [comment]
I believe we can also extend this to also use a worker node as part of the build automation with the following:

```
{
    "remote.SSH.remotePlatform": {
        "sharc.shef.ac.uk": "linux",
    },
    "terminal.integrated.shellArgs.linux": [ "-c","export FAF=FEF ; exec qrshx -V",]
    "terminal.integrated.automationShell.linux": [ "-c","export FAF=FEF ; exec qrshx -V",]
}
```

**Comment by jkwmoore** [comment]
> 
> 
> I believe we can also extend this to also use a worker node as part of the build automation with the following:
> 
> ```
> {
>     "remote.SSH.remotePlatform": {
>         "sharc.shef.ac.uk": "linux",
>     },
>     "terminal.integrated.shellArgs.linux": [ "-c","export FAF=FEF ; exec qrshx -V",]
>     "terminal.integrated.automationShell.linux": [ "-c","export FAF=FEF ; exec qrshx -V",]
> }
> ```

To amend this further - dropping to the login node is sub-optimal for many many reasons including people fork bombing themselves out of the cluster so the following is probably a better idea:

```
    "terminal.integrated.shellArgs.linux": [ "-c","export FAF=FEF ; exec qrshx -V || exit ",]
    "terminal.integrated.automationShell.linux": [ "-c","export FAF=FEF ; exec qrshx -V || exit ",]
```

tl;dr Either you get a worker session or you do not connect at all *not tested fully.

**Comment by IanSudbery** [comment]
This works to start an interactive shell on a worker, but still starts the server on the head node. 

The variable "terminal.integrated.automationShell.linux" is looking for a path to a shell (I think), not a set of shellArgs. The help entry reads:

> A path that when set will override terminal.integrated.shell.linux and ignore shellArgs values for automation-related terminal usage like tasks and debug.

Either way, it continues to start the server on the head node. Oh well.

**Comment by jkwmoore** [comment]
As per: https://code.visualstudio.com/docs/supporting/troubleshoot-terminal-launch

`terminal.integrated.automationShell.{platform} - Shell path for automation-related terminal usage like tasks and debug.`

In which case it sounds like one can simply substitute this with a path to their shell command to spawn an interactive session. 

e.g.  

`"terminal.integrated.automationShell.linux": [ "qrshx -V || exit ",]`

* not tested

**Comment by IanSudbery** [comment]
Unfortunately, this continues to run the server on the head node.

**Comment by jkwmoore** [comment]
It may work with `/usr/local/scripts/qrshx` ?

**Comment by jkwmoore** [comment]
On further investigation 
`
    "terminal.integrated.automationShell.linux": [ "-c","export FAF=FEF ; exec qrshx -V",]`

Should only be looking for a shell path e.g. /usr/bin/bash - I have not yet found which options will get automation shells (.e.g debugging) to spawn correctly on the cluster.

For now the following options are the most sane - denying access if a session off the login node is not found.

`
    "terminal.integrated.shellArgs.linux": [ "-c","exec qrshx -V || exit ",]
`

I have not extensively tested the above as 100% killing your session if on a login node however. 

Further login around cat /etc/hostname and if contains login kill session.
